src/otx/core/data/dataset/base.py

```diff
@@ -158,11 +166,24 @@ def _get_img_data_and_shape(self, img: Image) -> tuple[np.ndarray, tuple[int, in
             msg = "Cannot get image data"
             raise RuntimeError(msg)
 
-        img_data = self._cache_img(key=key, img_data=img_data.astype(np.uint8))
+        if roi:
+            # extract ROI from image
+            shape = roi["shape"]
+            h, w = img_data.shape[:2]
+            x1, y1, x2, y2 = (
```

Comment:
Could there be edge cases where the ROI is invalid? If so, could you add checks to validate the area and ensure that coordinates  x1, y1, x2, y2 are validated?

Reply:
These ROI can only come from Geti. I am not sure if there can be invalid bboxes. Of course, I can raise an error in case when (x2 < x1 or y2 < y1) (negative area). If you have other ideas how to handle this case, please, suggest!

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/4105/commits/f703e596013858fbfc9cc587f25a9b4579f50f9b